pandas/_libs/src/parser/tokenizer.c

```diff
@@ -1778,20 +1778,157 @@ double precise_xstrtod(const char *str, char **endptr, char decimal,
     return number;
 }
 
+/* copy a decimal number string in form `decimal` and `tsep` and `sci` as
+   decimal point, thousands separator and sci exponent character to a an
+   equivalent c-locale decimal string (striping tsep, replacing `decimal`
+   with '.'). Return NULL if nothing could be copied.
+*/
+
+char* str_copy_decimal_str_c(const char *s, char **endpos, char decimal,
```

Comment:
I think this would be a lot simpler if you just malloc a buffer, then copy from the original buffer into the new one ignoring the `thousands` separator and replacing the `decimal` separator with a period to satisfy the float constructor requirements

Reply:
The problem is that blind thousands separator removal could create strings that parse but should not. E.g. (with (decimal='.' and tsep='_') '1.2e_2' -> '1.2e2'.

Reply:
Can you add that as a test case? And noted, though could still simplify if you only do that up until the point that you've found the decimal separator